runtime/handler.go

```diff
@@ -194,8 +195,10 @@ func ForwardResponseMessage(ctx context.Context, mux *ServeMux, marshaler Marsha
 		w.Header().Set("Content-Length", strconv.Itoa(len(buf)))
 	}
 
-	if _, err = w.Write(buf); err != nil {
-		grpclog.Errorf("Failed to write response: %v", err)
+	if _, ok := resp.(*emptypb.Empty); !ok {
+		if _, err = w.Write(buf); err != nil {
+			grpclog.Errorf("Failed to write response: %v", err)
+		}
```

Comment:
I understand the problem, but I don't think this is the right conditional. The error is because we're trying to write a body when the request method doesn't allow it. Shouldn't we just check if `req.Method == http.MethodGet` or something like that?

Reply:
Thanks @johanbrandhorst for the review. 

The error occurs because status codes 204 and 304 do not allow a response body. 

Ideally, we should check if the response status is one of these codes before attempting to write a body. However, there isn’t a straightforward way to retrieve the status code from http.ResponseWriter after it has been written.

Also, req.Method == http.MethodGet alone is not sufficient, as a GET request can still return a non-empty response body. We could add a check for DELETE method calls, but again, there’s no guarantee that they will always have an empty response.

Reply:
OK, alternative idea. How about we intercept the error and just don't log it?
```suggestion
	if _, err = w.Write(buf); err != nil && !errors.Is(err, http.ErrBodyNotAllowed) {
		grpclog.Errorf("Failed to write response: %v", err)
	}
```

Reply:
Works for me, thank you. Addressed it here [ff571ca](https://github.com/grpc-ecosystem/grpc-gateway/pull/4902/commits/ff571ca2c37de587920d298138199eac6ddfed28)